libbeat/template/template.go

```diff
@@ -41,6 +45,39 @@ func New(beatVersion string, beatName string, esVersion string, config TemplateC
 		pattern = name + "-*"
 	}
 
+	compiler := func(key string, ops []fmtstr.VariableOp) (fmtstr.FormatEvaler, error) {
+		if key == "[beat.name]" {
+			return Formatter{s: beatName}, nil
+		}
+
+		if key == "[beat.version]" {
+			return Formatter{s: bV.String()}, nil
+		}
+
+		logp.Debug("template", "Lookup key for pattern not available: %s", key)
+		return nil, nil
```

Comment:
yes, will return an error if there is a pattern but doesn't match either of the two.